can not build when open microphone permission on iOS 14.6

## 🐛 Bug Report

can not build when open microphone permission

Podfile

```
post_install do |installer|
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)
    target.build_configurations.each do |config|
        config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '11.0'
        config.build_settings['ENABLE_BITCODE'] = 'NO'
          config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] ||= [
          '$(inherited)',
          'PERMISSION_MICROPHONE=1',
          'PERMISSION_CONTACTS=0',
          'PERMISSION_SPEECH_RECOGNIZER=0',
           'PERMISSION_MEDIA_LIBRARY=0',
           'PERMISSION_SENSORS=0'
        ]
        end
  end
end
```

output
```
Failed to build iOS app
Error output from Xcode build:
↳
    2021-06-28 15:14:00.574 xcodebuild[27997:3448308]  DVTAssertions: Warning in /Library/Caches/com.apple.xbs/Sources/DVTiOSFrameworks/DVTiOSFrameworks-18108/DTDeviceKitBase/DTDKRemoteDeviceData.m:371
    Details:  (null) deviceType from dc1fa74f1965dc3e7c4c693edc845dd91c7ec9a1 was NULL when -platform called.
    Object:   <DTDKMobileDeviceToken: 0x7fc5c653c920>
    Method:   -platform
    Thread:   <NSThread: 0x7fc5bed24210>{number = 2, name = (null)}
    Please file a bug at https://feedbackassistant.apple.com with this warning message and any useful information you can provide.
    ** BUILD FAILED **


Xcode's output:
↳
    /Users/miskolee/.pub-cache/hosted/pub.flutter-io.cn/permission_handler-8.1.1/ios/Classes/PermissionManager.m:75:59: warning: 'openURL:' is deprecated: first deprecated in iOS 10.0 [-Wdeprecated-declarations]
            BOOL success = [[UIApplication sharedApplication] openURL:[NSURL URLWithString:UIApplicationOpenSettingsURLString]];
                                                              ^~~~~~~
                                                              openURL:options:completionHandler:
    In module 'UIKit' imported from /Users/miskolee/develop/src/noiselabs/noise-audio-platform/flutter_plugins/noise_audio_platform_native/example/ios/Pods/Target Support Files/permission_handler/permission_handler-prefix.pch:2:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS14.5.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIApplication.h:123:1: note: 'openURL:' has been explicitly marked deprecated here
    - (BOOL)openURL:(NSURL*)url API_DEPRECATED_WITH_REPLACEMENT("openURL:options:completionHandler:", ios(2.0, 10.0)) NS_EXTENSION_UNAVAILABLE_IOS("");
    ^
    1 warning generated.

```



### Expected behavior

### Reproduction steps

### Configuration

**Version:**   8.1.1

**Platform:** 
 iOS 14.6 on iPad

--- a/permission_handler/CHANGELOG.md
+++ b/permission_handler/CHANGELOG.md
@@ -1,3 +1,7 @@
+## 8.1.2
+
+* Suppress deprecation warnings on iOS for code that specifically targets older iOS versions (see issue [#607](https://github.com/Baseflow/flutter-permission-handler/issues/607)). 
+
 ## 8.1.1
 
 * Fixed deprecation issue when checking phone capabilities on iOS (see issue [#597](https://github.com/Baseflow/flutter-permission-handler/issues/597)).

--- a/permission_handler/ios/Classes/PermissionManager.m
+++ b/permission_handler/ios/Classes/PermissionManager.m
@@ -72,8 +72,11 @@
                                      result([[NSNumber alloc] initWithBool:success]);
                                  }];
     } else if (@available(iOS 8.0, *)) {
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
         BOOL success = [[UIApplication sharedApplication] openURL:[NSURL URLWithString:UIApplicationOpenSettingsURLString]];
         result([[NSNumber alloc] initWithBool:success]);
+#pragma clang diagnostic pop
     } else {
         result(@false);    
     }

--- a/permission_handler/ios/Classes/strategies/ContactPermissionStrategy.m
+++ b/permission_handler/ios/Classes/strategies/ContactPermissionStrategy.m
@@ -47,6 +47,8 @@
         }
 
     } else {
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
         ABAuthorizationStatus status = ABAddressBookGetAuthorizationStatus();
 
         switch (status) {
@@ -58,6 +60,7 @@
                 return PermissionStatusPermanentlyDenied;
             case kABAuthorizationStatusAuthorized:
                 return PermissionStatusGranted;
+#pragma clang diagnostic pop
         }
     }
 
@@ -77,6 +80,8 @@
 }
 
 + (void)requestPermissionsFromAddressBook:(PermissionStatusHandler)completionHandler {
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
     ABAddressBookRequestAccessWithCompletion(ABAddressBookCreate(), ^(bool granted, CFErrorRef error) {
         if (granted) {
             completionHandler(PermissionStatusGranted);
@@ -84,6 +89,7 @@
             completionHandler(PermissionStatusPermanentlyDenied);
         }
     });
+#pragma clang diagnostic pop
 }
 @end
 

--- a/permission_handler/pubspec.yaml
+++ b/permission_handler/pubspec.yaml
@@ -1,6 +1,6 @@
 name: permission_handler
 description: Permission plugin for Flutter. This plugin provides a cross-platform (iOS, Android) API to request and check permissions.
-version: 8.1.1
+version: 8.1.2
 homepage: https://github.com/baseflowit/flutter-permission-handler
 
 flutter:

